### PR TITLE
CI: Start Testing on Python 3.11

### PR DESCRIPTION
**Pull request by lithomas1**
- [ ] xref #46680 (Replace xxxx with the Github issue number)
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

It's that time of the year again.

Surprisingly, there are not a lot of failures this year.

### CODE REVIEW

**Comment by EwoutH** [comment]
Thanks for the effort!

[Cython](https://github.com/cython/cython) 0.29.29 and 0.29.30 were released and [NumPy](https://github.com/numpy/numpy) bumped their Cython requirement on the main branch, so I think if you rebase this (and thereby trigger the CI again), it will come a bit further.

**Comment by lithomas1** [comment]
I think we already picked up 0.29.29 here(it compiles fine). I can try again with 0.29.30, but sadly these test failures look real.

File: pandas/tests/scalar/test_nat.py [hunk 1]
```diff
@@ -315,30 +310,31 @@ def test_overlap_public_nat_methods(klass, expected):
...
 
 @pytest.mark.parametrize(
-    "compare",
+    "klass",
     (
-        _get_overlap_public_nat_methods(Timestamp, True)
-        + _get_overlap_public_nat_methods(Timedelta, True)
+        Timestamp,
+        Timedelta
     ),
```
Comment (EwoutH) [inline pandas/tests/scalar/test_nat.py:317]: If you combine those four lines to a single line with `(Timestamp, Timedelta),`, it conforms to black formatting and the linting error will go away.

```diff
-    (
-        Timestamp,
-        Timedelta
-    ),
+    (Timestamp, Timedelta),
```

**Comment by EwoutH** [comment]
@lithomas1 Any progress on this effort? Should we open issues for the test failures?

**Comment by lithomas1** [comment]
@EwoutH No, they are mainly because an error message changed, and so should be relatively easy to fix. Does anyone downstream need this(can prioritize this if so)? 

I'm a bit short on time right now, so I'll be squeezing in commits here and there, but it might take me a while to fix up all the tests. 

You might also want to hold off reviewing right now, as there are going to be a lot of WIP commits, and I don't want to waste your time reviewing. I can ping you when this is ready.

**Comment by lithomas1** [comment]
@github-actions pre-commit

**Comment by lithomas1** [comment]
@github-actions pre-commit

**Comment by EwoutH** [comment]
@lithomas1 Thanks for the effort so far. Are you still working on this?

**Comment by lithomas1** [comment]
This PR is currently superseded by #47442. That PR is still blocked by #47467. I expect to finish up #47467 and get it merged within the next week.

**Pull request closed without merging**